recipes/beyla-golden-signals/gce/beyla-config.yaml

```diff
@@ -0,0 +1,45 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Beyla can be configured to listen to applications running on multiple ports.
+# Beyla can also be configured globally using environment variables.
+# Environment variables have priority over the properties in the config file.
+open_port: 8080,8082,8083
```

Comment:
Does this limit beyla to only those ports?  If so, we probably don't want to do that.

Reply:
`open_port` is a required configuration and cannot be omitted. We can **also** specify a range of ports (1-65536) to instrument all applications running on the VM owning any ports within the specified. 

Instrumenting all process might lead to generating noisy instrumentation though. Why do we not want to limit to specific ports for this sample though ?

Reply:
Ideally, a user should be able to follow these instructions with _their_ application, and it should just work.  We can provide instructions on how to reduce noise by adding the filter, but the goal is to provide something that works without any configuration, even if it isn't perfect.

Reply:
If the goal is to provide a Beyla config file that *works* with any user application, then I believe its ok to put a wide range that covers all ports (1-65536). But seeing as they would most likely a user would want to edit this file (*service-discovery section*), I would say instead of doing (1-65536), we should do something like: `80, 443, (8080-8999)`.  These are most common user application ports and would work with most use-cases.

This would also indicate to the user that they could configure the file with a mix of specific ports and ranges.

Reply:
I don't know enough about how people use ports.  I'm fine with either option.  Are there any ports that are particularly noisy if you enable all ports?

Reply:
So I enabled instrumentation on all ports, it wasn't terrible as the number of metrics being reported did not increase by a lot.
I noticed: 
 - slight increase in the memory usage (virtual memory) of the Beyla process (1.23 GiB to 1.36 GiB)
 - huge increase in the CPU usage of the Beyla process (1.62 ms/s &rarr; 6.2 ms/s)

Reply:
Adding a comment works for me.